File: Submissions/917_Reverse_Only_Letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string S) {
        for(int i=0, j=S.length()-1; j>i;){
            if(!isLetter(S[i])) {
                ++i;
                continue;
            }
            if(!isLetter(S[j])) {
                --j;
                continue;
            }
            swap(S[i], S[j]);
            ++i, --j;
        }
        return S;
    }
    
    bool isLetter(const char ch){
        return (ch >='a' && ch<='z') || (ch >='A' && ch <='Z'); 
    }
};

File: Kotlin/917_Reverse_Only_Letters.kt
class Solution {
    fun reverseOnlyLetters(s: String): String {
        val result = s.toCharArray()
        var i = 0
        var j = s.length - 1
        
        while (i < j) {
            if(!result[i].isLetter()) {
                ++i
                continue
            }
            else if (!result[j].isLetter()){
                --j
                continue
            }
            swap(result, i, j);
            ++i
            --j
        }
    
        return String(result);
    }
    
    private fun swap(c: CharArray, i: Int, j: Int) {
        val temp = c[i]
        c[i] = c[j]
        c[j] = temp
    }
}